client/app/components/sortable/index.js

```diff
@@ -0,0 +1,80 @@
+import { isFunction, wrap } from 'lodash';
+import React, { useRef, useState } from 'react';
+import PropTypes from 'prop-types';
+import cx from 'classnames';
+import { sortableContainer, sortableElement, sortableHandle } from 'react-sortable-hoc';
+
+import './style.less';
+
+export const DragHandle = sortableHandle(({ className, ...restProps }) => (
+  <div className={cx('drag-handle', className)} {...restProps} />
+));
+
+export const SortableContainerWrapper = sortableContainer(({ children }) => children);
+
+export const SortableElement = sortableElement(({ children }) => children);
+
+export function SortableContainer({ disabled, containerProps, children, ...wrapperProps }) {
+  const containerRef = useRef();
+  const [isDragging, setIsDragging] = useState(false);
+
+  wrapperProps = { ...wrapperProps };
+  containerProps = { ...containerProps };
+
+  if (disabled) {
+    // Disabled state:
+    // - forbid drag'n'drop (and therefore no need to hook events
+    // - don't override anything on container element
+    wrapperProps.shouldCancelStart = () => true;
+  } else {
+    // Enabled state:
+
+    // - use container element as a default helper element
+    wrapperProps.helperContainer = wrap(wrapperProps.helperContainer, helperContainer => (
+      isFunction(helperContainer) ?
+        helperContainer(containerRef.current) :
+        containerRef.current
+    ));
+
+    // - hook drag start/end events
+    wrapperProps.updateBeforeSortStart = wrap(wrapperProps.updateBeforeSortStart, (updateBeforeSortStart, ...args) => {
+      setIsDragging(true);
+      if (isFunction(updateBeforeSortStart)) {
+        updateBeforeSortStart(...args);
+      }
+    });
+    wrapperProps.onSortEnd = wrap(wrapperProps.onSortEnd, (onSortEnd, ...args) => {
+      setIsDragging(false);
+      if (isFunction(onSortEnd)) {
+        onSortEnd(...args);
+      }
+    });
+
+    // - update container element: add classes and take a ref
+    containerProps.className = cx(
+      'sortable-container',
+      { 'sortable-container-dragging': isDragging },
```

Comment:
Why not a `data-dragging` modifier instead? I thought it was becoming a standard :slightly_smiling_face:

Reply:
@gabrieldutra +1

Reply:
I think because we don't use `data-` attributes anywhere except of `data-test`. Also, IMHO class-based selectors are easier to read in CSS files

Reply:
I use it all over the place  😬

Imo, as long as `data-prop1={condition || null}`, the selector is as coherent in the CSS

```css
[data-prop1] {
  /* rule */
}
```

Reply:
Well, I think we need some convention for this :grin: I personally don't like (and avoid to use) attribute-based selectors for few reasons:
- performance (all browsers have some optimizations for id/class-based selectors, but don't optimize tag/attribute based selectors)
- attributes selectors have a least specificity, so they're very easy to override (especially accidentally) by classes, but almost no way to override other rules by attribute-based selector. - (as a part of previous one) `class` attribute is a list and may contain multiple values (the next class overrides rules of previous. Data attributes selectors usually are used by 1) presense 2) exact value - so classes are more flexible.

I feel comfortable to use attributes for things that are standalone and "boolean" in their nature (like `disabled` property). In this case, we actually have two flags: `draggable` which defines some styles for draggable _container_ (another thing that feels wrong to me - `draggable` flag should be set for element which is draggable itself), and `dragged` which depends on and overrides some styles of `draggable` - so actually these two attributes were used as classes - when second one extends/overrides previous one.

Actually it's quite a good topic for discussion :thinking:

Reply:
I think these are all good points.
I personally have not encountered these potential downsides, but I appreciate the thoughtfulness and am ok with the approach.

Reply:
:thinking: personally I prefer the data attributes, but I haven't taken the opportunity to answer your arguments yet.

We should find a convention, but this shouldn't block this PR.